File: 2thread/src/main/java/com/example/a2thread/MainActivity.java
package com.example.a2thread;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TestThread mThread = new TestThread();
        mThread.start();

        TestRunnable runnable = new TestRunnable();
        Thread mThread2 = new Thread(runnable);
        mThread2.start();
    }
}

File: kotlin/src/main/java/com/example/kotlin/MainActivity.kt
package com.example.kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch
import java.util.concurrent.TimeUnit

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        addNumber(1)


        GlobalScope.launch {
            println("codes run in coutines scope")

            delay(15000)

            println("code run in courtine scope finished")
        }


    }
}